components/App.vue

```diff
@@ -196,15 +198,34 @@ export default {
           this.$store.commit('clearPreview');
         }
 
-        console.log("Buy event processed.");
+        console.log("[KetherHomepage] Buy event processed.");
       }.bind(this));
 
       contract.on('Publish', function(idx, link, image, title, NSFW) {
-        this.$store.commit('addAd', {idx: idx.toNumber(), link, image, title, NSFW});
-        console.log("Publish event processed.");
+        this.$store.commit('updateAd', {idx: idx.toNumber(), toUpdate: {link, image, title, NSFW}});
+        console.log("[KetherHomepage] Publish event processed.");
+      }.bind(this));
+
+      contract.on('SetAdOwner', function(idx, _from, to) {
```

Comment:
This is going to collide with initState event-based loading which also subs to this event.

Reply:
Fixed now.